google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerTableExporter.java

```diff
@@ -55,47 +60,90 @@ public SpannerTableExporter(SpannerDialect spannerDialect) {
+    // Use only the replaced commands if the list contains both normal InitCommands and
+    // ReplaceInitCommands.
+    if (initCommands.stream().anyMatch(cmd -> cmd instanceof ReplaceInitCommand)
+        && initCommands.stream().anyMatch(cmd -> !(cmd instanceof ReplaceInitCommand))) {
+      initCommands =
+          initCommands.stream()
+              .filter(cmd -> cmd instanceof ReplaceInitCommand)
+              .collect(Collectors.toList());
+    } else if (initCommands.stream().anyMatch(cmd -> cmd instanceof ReplaceInitCommand)) {
+      // Only ReplaceInitCommands, but there is nothing to replace, so we return early.
+      return;
+    }
+    for (InitCommand initCommand : initCommands) {
+      addStatementAfterDdlBatch(metadata, initCommand.getInitCommands());
+    }
   }
 
   @Override
   public String[] getSqlDropStrings(Table currentTable, Metadata metadata) {
     initializeUniqueConstraints(currentTable);
-    return buildSqlStrings(currentTable, metadata, Action.DROP);
+    return buildSqlStrings(currentTable, metadata, Action.DROP).toArray(new String[0]);
   }
 
   /**
    * Initializes the table exporter for if a new create-table or drop-table sequence is starting.
    */
   public void init(
-      Metadata metadata,
-      SpannerDatabaseInfo spannerDatabaseInfo,
-      Action schemaAction) {
+      Metadata metadata, SpannerDatabaseInfo spannerDatabaseInfo, Action schemaAction) {
     tableDependencyTracker.initializeDependencies(metadata, schemaAction);
     spannerTableStatements.initializeSpannerDatabaseInfo(spannerDatabaseInfo);
   }
 
-  private String[] buildSqlStrings(Table currentTable, Metadata metadata, Action schemaAction) {
+  private List<String> buildSqlStrings(Table currentTable, Metadata metadata, Action schemaAction) {
     Collection<Table> tablesToProcess = tableDependencyTracker.getDependentTables(currentTable);
 
-    List<String> ddlStatements = tablesToProcess.stream()
-        .flatMap(table -> {
-          if (schemaAction == Action.CREATE) {
-            return spannerTableStatements.createTable(table, metadata).stream();
-          } else {
-            return spannerTableStatements.dropTable(table).stream();
-          }
-        })
+    return tablesToProcess.stream()
+        .flatMap(
+            table -> {
+              if (schemaAction == Action.CREATE) {
+                return spannerTableStatements.createTable(table, metadata).stream();
+              } else {
+                return spannerTableStatements.dropTable(table).stream();
+              }
+            })
         .collect(Collectors.toList());
+  }
+
+  static void addStatementAfterDdlBatch(Metadata metadata, String[] statements) {
+    // Find the RunBatchDdl auxiliary object which can run statements after the DDL batch.
+    Optional<RunBatchDdl> runBatchDdl =
+        metadata.getDatabase().getAuxiliaryDatabaseObjects().stream()
+            .filter(obj -> obj instanceof RunBatchDdl)
```

Comment:
Just a recommendation: you can replace lambda method reference here: `RunBatchDdl.class::isInstance`.